internal/cluster/transition.go

```diff
@@ -402,6 +402,12 @@ func (th *transitionHandler) PostRefreshCluster(reason string) stateswitch.PostT
 			err            error
 			updatedCluster *common.Cluster
 		)
+
+		//if no changes need to be made return
+		if sCluster.srcState == swag.StringValue(sCluster.cluster.Status) && ok && reason == swag.StringValue(sCluster.cluster.StatusInfo) {
```

Comment:
What about setPendingUserResetIfNeeded function? May it be skipped if these fields have not changed?